### PR TITLE
PLANET-6432: Add reading time to post details

### PR DESCRIPTION
**Pull request by lithrel**
Ref: https://jira.greenpeace.org/browse/PLANET-6432

Cf. https://github.com/greenpeace/planet4-master-theme/pull/1542

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [phobos](https://www-dev.greenpeace.org/test-phobos) | [admin](https://www-dev.greenpeace.org/test-phobos/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-phobos/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-phobos) | [composer-local.json](https://github.com/greenpeace/planet4-test-phobos/blob/main/composer-local.json)

:watch: 2021.12.13 08:33:16

**Review by GP-Dan-Tovbein — COMMENTED** [review]
I was testing this feature and works well, good job! However, I detected an inconsistent behaviour and I would like your opinion.

I know that the reading time is calculated by length of words, thus, I've created 2 different pages with the same content, but in one of them I removed all spaces. (this is a very special case LOL)

[With spaces](https://www-dev.greenpeace.org/test-pandora/blogs/49626/reading-time-demo-with-spaces/): 19 min
[Without spaces](https://www-dev.greenpeace.org/test-pandora/blogs/49630/reading-time-demo-without-spaces/): 2 min

What do you think?

**Comment by lithrel** [comment]
> What do you think?

That's an interesting test :grin:  I can see a few solutions for that, where we adjust the score with the actual total length of the text; but the [ICU](https://icu.unicode.org/) lib that is used is made for a real language usage, with [well defined word/sentence separators, punctuation, etc.](https://github.com/unicode-org/icu/blob/49dda34fb175240a7724c7e039a270126ff7d900/icu4c/source/data/locales/fr.txt), depending on the locale selected. So I don't think it's necessary to handle this case for now, we can work on it if an editor has a real use case. And if we end up writing a manual override of the read time by post, this would probably become irrelevant.

**Comment by Inwerpsel** [comment]
> > What do you think?
> 
> That's an interesting test grin I can see a few solutions for that, where we adjust the score with the actual total length of the text; but the [ICU](https://icu.unicode.org/) lib that is used is made for a real language usage, with [well defined word/sentence separators, punctuation, etc.](https://github.com/unicode-org/icu/blob/49dda34fb175240a7724c7e039a270126ff7d900/icu4c/source/data/locales/fr.txt), depending on the locale selected. So I don't think it's necessary to handle this case for now, we can work on it if an editor has a real use case. And if we end up implementing a manual override of the read time by post, this would probably become irrelevant.

It's also something that struck me when investigating the reading time (see [relevant section in doc](https://docs.google.com/document/d/1D3dUrNwNBKUCyYXHSytd8fg-3b7QdV2SvmFnSI592mE/edit#heading=h.b4cgjfnpd4o9)), as I was familiar with WPM from typing speed which actually does count the characters. But it provides a good approximation in almost all cases.

**Comment by Inwerpsel** [comment]
I guess the submodule update is in here by accident?

File: assets/src/blocks/Articles/ArticlePreview.js [hunk 1]
```diff
@@ -103,6 +104,7 @@ export class ArticlePreview extends Component {
         post_title,
         post_excerpt,
         date_formatted,
+        reading_time
```
Comment (GP-Dan-Tovbein) [inline assets/src/blocks/Articles/ArticlePreview.js:107]: Small detail.. missing the `,` at the EOL

**Pull request merged**